File: codeWars/JavaScript/maxProfit.js
function getMostProfitFromStockQuotes(quotes) {

  let sellPrice = Math.max(...quotes);
  let profit = 0;

  while(quotes.length > 0){
    
    profit += sellPrice - quotes[0];
    quotes.shift();

    sellPrice = Math.max(...quotes);
  }

  return profit;
}

getMostProfitFromStockQuotes([ 1, 2, 3, 4, 5, 6 ]);
getMostProfitFromStockQuotes([ 6, 5, 4, 3, 2, 1 ]);
getMostProfitFromStockQuotes([ 1, 2, 10, 3, 2, 7, 3, 2 ]);

File: codeWars/Python/max_profit.py
import numpy as np

def get_most_profit_from_stock_quotes(quotes): 
  # your code here
  np_arr = np.array(quotes)
  sell_price = np.amax(np_arr)
  profit = 0
    
  while len(np_arr) > 0:
    
    current_price = np_arr[0]
    
    profit += sell_price - current_price
    
    quotes.pop(0)
    
    np_arr = np.array(quotes)
    
    sell_price = np.array(np_arr)
  
  print(quotes)
  
get_most_profit_from_stock_quotes([1, 2, 10, 3, 2, 7, 3, 2])